### ISSUE TITLE
Assignments Sidebar

### ISSUE THREAD

**Issue by quick007** [CoursifyStudios/Coursify#256]
Revamp the assignment sidebar for teachers and students. 

tl;dr:

- Adds sorting, filtering and searching
- By default, assignments are categorized for a user (unless the user is searching)
- Shows more relevant info to the student
- Default sorting, relevancy, prioritizes assignments that are due today or tomorrow that haven't been submitted yet.

## New Assignment Preview

For students, the preview won't change much. The only difference will be nothing submitted, draft started, submitted, and hidden icons rather than the checkmark. Students can select "hidden" to remove it from their feed and will show "hidden from view" on the teacher's end".

Assignments will be grouped into (all names tba) Due Later On (due in the future, default closed), Due Imminently (due today not past due date, due tmrw; if weekend fetched until Monday or today and tmrw, whichever is longer), Late (all late assignments), Comepleted (all other assignments)

The teachers on the other hand will see radical changes to their UI. Heres a mockup:

![Image](https://github.com/CoursifyStudios/Coursify/assets/69743171/24bff00d-d7a8-4996-a807-47d9414089ff)

Note that the scores part at the bottom will be divided into the types of grades students not, with the average/median/average middle 50 being options to change to.

Late/resubmitted will show the students that resubmitted/are late in the preview

## Top Bit

The top bit will include sorting, filters, and a search bar (not pictured)

![Image](https://github.com/CoursifyStudios/Coursify/assets/69743171/17b21f41-7336-4502-bda1-8a1bc7033100)

### Sorting

Users will be able to sort by due date, publish date, and name with ascending/descending options. They will also be able to use the default sorting called relevancy. It prioritizes assignments that are due today or tomorrow that haven't been submitted yet.

### Filters
Students will be able to filter by:

true/false: not submitted, draft, submitted, hidden, 

select/type: assignment type, class, search query

### Search

Search will have a separate search bar, but act pretty much like a filter

## Pagination

Pagination will be included with this pr as well
